### PR TITLE
Support for hidding elements

### PR DESCRIPTION
**Pull request by bell07**
The next PR like the last one. It is a basic functionality of dynamic screens. If you try the demo application, using the toggle the text field will be hidden and shown again. Note the content of the text field is not lost by hiding.

### CODE REVIEW

File: docs/API.md [hunk 1]
```diff
@@ -58,6 +58,8 @@
 * element:getSize() - get the size
 * element:setBackground(image) - Set the background of element. Please note a size needs to be defined on element
 * element:getBackground() - get the current background
+* element:setIsHidden(bool) - set the hidden status (hide=true, unhide=false)
```
Comment (rubenwardy) [inline docs/API.md]: I'd prefer `setVisible`

Comment (bell07) [inline docs/API.md]: You are right. What would I do without you? I'll change it. ;)

**Comment by bell07** [comment]
Done. Now it is get/setVisible, true by default.

File: smartfs.lua [hunk 2]
```diff
@@ -304,7 +304,9 @@ function smartfs._makeState_(form, newplayer, params, is_inv, nodepos)
 				res = "size["..self._size.w..","..self._size.h.."]"
 			end
 			for key,val in pairs(self._ele) do
-				res = res .. val:getBackgroundString() .. val:build()
+				if val:getVisible() == true then
```
Comment (rubenwardy) [inline smartfs.lua]: `== true` is redundant

File: docs/example.lua [hunk 3]
```diff
@@ -10,7 +10,15 @@ local s = smartfs.create("smartfs:form", function(state)
 	usr:setBackground("halo.png")
 	state:field(7.25,1.25,3,1,"txt","Textbox")
 	state:image(0,0,2,2,"img","default_stone.png")
-	state:toggle(0,2,3,1,"tg",{"plenty..","of..","custom..","elements"})
+	local toggle = state:toggle(0,2,3,1,"tg",{"plenty..","of..","custom..","elements"})
+	toggle:onToggle(function(self, state, player)
+		if state:get("ta"):getVisible() == false then
```
Comment (rubenwardy) [inline docs/example.lua]: `state:get("ta"):setVisible(not state:get("ta"):getVisible())`

also, I forget, but isn't `self == state:get("ta")`? If so:

`self:setVisible(not self:getVisible())`

Comment (bell07) [inline docs/example.lua]: The sample works as expected. If hidden then show up, else hide. The first "getVisible" is the check if the "ta" field is hidden.
About usage of local references: It is a sample how the smartfs could be used. If I need the reference more then 1-2 times you are right, usage of local reference is better. But using of local references are already shown multiple times, like local usr, local toggle... The new sample is a showcase for the get() method too ;)

**Pull request merged**